lib/services/saml_idp_service_provider.go

```diff
@@ -121,15 +123,68 @@ func GenerateIdPServiceProviderFromFields(name string, entityDescriptor string)
 	return &s, nil
 }
 
+// supportedACSBindings is the set of AssertionConsumerService bindings that teleport supports.
+var supportedACSBindings = map[string]struct{}{
+	saml.HTTPPostBinding:     {},
+	saml.HTTPRedirectBinding: {},
+}
+
+// ValidateAssertionConsumerService checks if a given assertion consumer service is usable by teleport. Note that
+// it is permissible for a service provider to include acs endpoints that are not compatible with teleport, so long
+// as at least one _is_ compatible.
+func ValidateAssertionConsumerService(acs saml.IndexedEndpoint) error {
+	if _, ok := supportedACSBindings[acs.Binding]; !ok {
+		return trace.BadParameter("unsupported acs binding: %q", acs.Binding)
+	}
+
+	if acs.Location == "" {
+		return trace.BadParameter("acs location endpoint is missing or could not be decoded for %q binding", acs.Binding)
+	}
+
+	return trace.Wrap(ValidateAssertionConsumerServicesEndpoint(acs.Location))
+}
+
+// FilterSAMLEntityDescriptor performs a filter in place to remove unsupported and/or insecure fields from
+// a saml entity descriptor. Specifically, it removes acs endpoints that are either of an unsupported kind,
+// or are using a non-https endpoint. We perform filtering rather than outright rejection because it is generally
+// expected that a service provider will successfully support a given ACS so long as they have at least one
+// compatible binding.
+func FilterSAMLEntityDescriptor(ed *saml.EntityDescriptor) error {
+	var originalCount int
+	var filteredCount int
+	for i := range ed.SPSSODescriptors {
+		filtered := ed.SPSSODescriptors[i].AssertionConsumerServices[:0]
```

Comment:
Took me a good couple of minutes to convince myself that this is okay.

Consider 
```
filtered := slices.DeleteFunc(
    ed.SPSSODescriptors[i].AssertionConsumerServices,
    func(acs ACS) bool { return ValidateAssertionConsumerService(acs) != nil })
```